src/Synapse/TestHelper/WebTestCase.php

```diff
@@ -0,0 +1,180 @@
+<?php
+namespace Synapse\TestHelper;
+
+use Application;
+use Silex\Provider\SessionServiceProvider;
+use Silex\WebTestCase as WebCase;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernel;
+use Synapse;
+
+class WebTestCase extends WebCase
+{
+    protected $mocks = [];
+
+    protected $authToken = [
+        'client_id' => 123,
+        'user_id'   => 42,
+        'scope'     => '',
+    ];
+
+    public function setUp()
+    {
+        // Don't call createApplication by default
+    }
+
+    public function setMocks(array $mocks)
+    {
+        foreach ($mocks as $alias => $class) {
+            $this->mocks[$alias] = $this
+                ->getMockBuilder($class)
+                ->disableOriginalConstructor()
+                ->getMock();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createClient(array $server = array())
+    {
+        return new WebTestClient($this->app, $server);
+    }
+
+    public function createApplication()
+    {
+        throw new \Exception('Use createApplicationWithServices instead');
+    }
+
+    /**
+     * Creates the application.
+     *
+     * @return Synapse\Application
+     */
+    public function createApplicationWithServices(array $services)
+    {
+        defined('WEBDIR') or define('WEBDIR', realpath(__DIR__));
+        defined('APPDIR') or define('APPDIR', realpath(WEBDIR.'/..'));
+        defined('DATADIR') or define('DATADIR', APPDIR.'/data');
+        defined('TMPDIR') or define('TMPDIR', '/tmp');
+        date_default_timezone_set('UTC');
+
+        $applicationInitializer = new Synapse\ApplicationInitializer;
+
+        $app = $applicationInitializer->initialize();
+
+        $defaultServices = new Synapse\Application\Services;
```

Comment:
I'm thinking there is probably a reduced set of providers we can use...for example, we won't need a db for these things.